pyomo/util/latex_printer.py

```diff
@@ -0,0 +1,1484 @@
+
+
+def handle_negation_node(visitor, node, arg1):
+    arg1, tsh = precedenceChecker(node, arg1)
+    return '-' + arg1
+
+
+def handle_product_node(visitor, node, arg1, arg2):
+    arg1, arg2 = precedenceChecker(node, arg1, arg2)
+    return ' '.join([arg1, arg2])
+
+
+def handle_pow_node(visitor, node, arg1, arg2):
+    arg1, arg2 = precedenceChecker(node, arg1, arg2)
+    return "%s^{%s}" % (arg1, arg2)
+
+
+def handle_division_node(visitor, node, arg1, arg2):
+    return '\\frac{%s}{%s}' % (arg1, arg2)
+
+
+def handle_abs_node(visitor, node, arg1):
+    return ' \\left| ' + arg1 + ' \\right| '
+
+
+def handle_unary_node(visitor, node, arg1):
+    fcn_handle = node.getname()
+    if fcn_handle == 'log10':
+        fcn_handle = 'log_{10}'
+
+    if fcn_handle == 'sqrt':
+        return '\\sqrt { ' + arg1 + ' }'
+    else:
+        return '\\' + fcn_handle + ' \\left( ' + arg1 + ' \\right) '
+
+
+def handle_equality_node(visitor, node, arg1, arg2):
+    return arg1 + ' = ' + arg2
+
+
+def handle_inequality_node(visitor, node, arg1, arg2):
+    return arg1 + ' \\leq ' + arg2
+
+
+def handle_var_node(visitor, node):
+    return visitor.variableMap[node]
+
+
+def handle_num_node(visitor, node):
+    if isinstance(node, float):
+        if node.is_integer():
+            node = int(node)
+    return str(node)
+
+
+def handle_sumExpression_node(visitor, node, *args):
+    rstr = args[0]
+    for i in range(1, len(args)):
+        if args[i][0] == '-':
+            rstr += ' - ' + args[i][1:]
+        else:
+            rstr += ' + ' + args[i]
+    return rstr
+
+
+def handle_monomialTermExpression_node(visitor, node, arg1, arg2):
+    if arg1 == '1':
+        return arg2
+    elif arg1 == '-1':
+        return '-' + arg2
+    else:
+        return arg1 + ' ' + arg2
+
+
+def handle_named_expression_node(visitor, node, arg1):
+    # needed to preserve consistencency with the exitNode function call
+    # prevents the need to type check in the exitNode function
+    return arg1
+
+
+def handle_ranged_inequality_node(visitor, node, arg1, arg2, arg3):
+    return arg1 + ' \\leq ' + arg2 + ' \\leq ' + arg3
+
+
+def handle_exprif_node(visitor, node, arg1, arg2, arg3):
+    return 'f_{\\text{exprIf}}(' + arg1 + ',' + arg2 + ',' + arg3 + ')'
+
+    ## Could be handled in the future using cases or similar
+
+    ## Raises not implemented error
+    # raise NotImplementedError('Expr_if objects not supported by the Latex Printer')
+
+    ## Puts cases in a bracketed matrix
+    # pstr = ''
+    # pstr += '\\begin{Bmatrix} '
+    # pstr += arg2 + ' , & ' + arg1 + '\\\\ '
+    # pstr += arg3 + ' , & \\text{otherwise}' + '\\\\ '
+    # pstr += '\\end{Bmatrix}'
+    # return pstr
+
+
+def handle_external_function_node(visitor, node, *args):
+    pstr = ''
+    pstr += 'f('
+    for i in range(0, len(args) - 1):
+        pstr += args[i]
+        if i <= len(args) - 3:
+            pstr += ','
+        else:
+            pstr += ')'
+    return pstr
```

Comment:
This is a bit of a pain, but I can do it

Reply:
After evaluation, I agree with the sentiment, but do not believe this meets the standard of effort vs. reward.  It's a fairly involved change for something of questionable value.  I'll add it to my personal issue list, but will not implement it for the November pyomo release

Reply:
Sorry to push back, but my concern here is this is borderline on the outputting something inaccurate front. If a bunch of functions are named `f`, I would expect them to be the same function, and they aren't necessarily. I think throwing a `MouseTrap` for all external functions would be fine at this point, if you want to keep it simple. But naming them all `f` is what concerns me a bit.

Reply:
So an easy thing is I can basically have a counter and do f_1, f_2, f_3...  That I can implement easily.  But adding to the component map of latex names is not trivial (I have to detect all the black boxes, do substitutions, map back to the desired names).  If we felt this was a significant gain (or if I had infinite bandwidth) then this would be a no brainer, but I'm not sure it is worth the time. I will implement the counter strategy

Reply:
The counter strategy sounds good to me!